### PR TITLE
msg.sender should not exist on Solana

### PR DESCRIPTION
**Pull request by seanyoung**
Explanation in `docs/language/function.rst` 

- Remove  `msg.sender`
- Remove `caller` on yul

### CODE REVIEW

File: docs/language/functions.rst [hunk 1]
```diff
@@ -217,13 +217,19 @@ Function modifiers are used to check pre-conditions or post-conditions for a fun
...
-.. include:: ../examples/function_modifier.sol
+.. include:: ../examples/substrate/function_modifier.sol
   :code: solidity
 
 The function `foo` can only be run by the owner of the contract, else the ``require()`` in its
 modifier will fail. The special symbol ``_;`` will be replaced by body of the function. In fact,
 if you specify ``_;`` twice, the function will execute twice, which might not be a good idea.
 
+On Solana, ``msg.sender`` does not exist, so the usual way to implement a similar test is using
+an `authority` accounts rather than an owner account.
```
Comment (LucasSte) [inline docs/language/functions.rst:228]: I'm starting to think that the code blocks on our documentation should have one tab for each target. This is not something for this PR, though.

File: docs/targets/solana.rst [hunk 2]
```diff
@@ -183,6 +183,25 @@ this. The balance of an account can be credited without any code being executed.
 
 ``receive()`` functions are not permitted on the Solana target.
 
+.. _msg_sender_solana:
+
+``msg.sender`` not available on Solana
+______________________________________
+
+On Ethereum, ``msg.sender`` is used to identify the account which either submitted
```
Comment (LucasSte) [inline docs/targets/solana.rst:191]:
```suggestion
On Ethereum, ``msg.sender`` is used to identify either the account that submitted
```
This is a restrictive clause.

File: docs/targets/solana.rst [hunk 3]
```diff
@@ -183,6 +183,25 @@ this. The balance of an account can be credited without any code being executed.
...
+the transaction, or the caller when one contract calls another. On Ethereum, each
+contract execution involves a single contract. On Solana, each contract execution
```
Comment (LucasSte) [inline docs/targets/solana.rst:193]: `contract execution involves a single contract`
Is this what you wanted to say?

File: docs/targets/solana.rst [hunk 4]
```diff
@@ -183,6 +183,25 @@ this. The balance of an account can be credited without any code being executed.
...
+involves many accounts, so there is not a single account which can be identified
+as a caller. So, the Solana VM has no mechanism for fetching the caller account.
+This means there is no way to implement ``msg.sender``.
+
+The way to implement this on Solana is to have an authority account for the contract,
+and this account must be a signer for the transaction (note that on Solana there
```
Comment (LucasSte) [inline docs/targets/solana.rst:199]:
```suggestion
that must be a signer for the transaction (note that on Solana there
```

**Review by LucasSte — CHANGES_REQUESTED** [review]

File: docs/language/functions.rst (hunk 1 above)
Comment (seanyoung) [reply docs/language/functions.rst:228]: What do you mean by tab?

Comment (xermicus) [reply docs/language/functions.rst:232]: I think he means [something like this](https://global.discourse-cdn.com/standard10/uploads/gohugo/original/2X/f/f4ff340603de07622145cbcc70e9b8cbf316f646.png) but for our targets. I had a similar idea a while ago, basically the same but putting the documentation text itself inside tabs, wherever there are differences between the targets. But I couldn't figure out how to do in RST.

**Review by LucasSte — APPROVED** [review]

File: docs/targets/solana.rst [hunk 5]
```diff
@@ -184,6 +184,26 @@ this. The balance of an account can be credited without any code being executed.
...
 
+.. _msg_sender_solana:
+
+``msg.sender`` not available on Solana
+______________________________________
+
+On Ethereum, ``msg.sender`` is used to identify either the account that submitted
+the transaction, or the caller when one contract calls another. On Ethereum, each
+contract execution can only use a single account, which provides the code and data.
+On Solana, each contract execution uses many accounts, so in many cases there is
```
[the remark this answers is not in the document]
Comment (seanyoung) [reply docs/targets/solana.rst:195]: Many accounts

[the remark this answers is not in the document]
Comment (seanyoung) [reply docs/targets/solana.rst:195]: Granted, Solidity executions use only two accounts for now - but that's not true for non-solidity callers. I should explain this better.

**Pull request merged**